From: Petr Wozniak <petr.wozniak@gmail.com>
Subject: [PATCH] uboot-mediatek: BPI-R4: boot from NVMe in the SNAND environment

--- a/defenvs/bananapi_bpi-r4_snand_env
+++ b/defenvs/bananapi_bpi-r4_snand_env
@@ -4,7 +4,7 @@
 bootargs=console=ttyS0,115200n1 pci=pcie_bus_perf root=/dev/fit0 rootwait ubi.block=0,fit
 bootconf=config-mt7988a-bananapi-bpi-r4
 bootconf_extra=mt7988a-bananapi-bpi-r4-spim-nand#mt7988a-bananapi-bpi-r4-emmc
-bootcmd=if pstore check ; then run boot_recovery ; else run boot_ubi ; fi
+bootcmd=if env exists nvme_boot ; then run boot_nvme ; else if pstore check ; then run boot_recovery ; else run boot_ubi ; fi ; fi
 bootdelay=0
 bootfile=openwrt-mediatek-filogic-bananapi_bpi-r4-initramfs-recovery.itb
 bootfile_bl2=openwrt-mediatek-filogic-bananapi_bpi-r4-snand-preloader.bin
@@ -15,7 +15,7 @@
 bootmenu_confirm_return=askenv - Press ENTER to return to menu ; bootmenu 60
 bootmenu_default=0
 bootmenu_delay=0
-bootmenu_title=      [0;34m( ( ( [1;39mOpenWrt[0;34m ) ) )  [0;36m[SPI-NAND][0m
+bootmenu_title=      [0;34m( ( ( [1;39mOpenWrt[0;34m ) ) )  [0;36m[SPI-NAND][0m
 bootmenu_0=Initialize environment.=run _firstboot
 bootmenu_0d=Run default boot command.=run boot_default
 bootmenu_1=Boot system via TFTP.=run boot_tftp ; run bootmenu_confirm_return
@@ -23,16 +23,18 @@
 bootmenu_3=Boot recovery system from NAND.=run boot_recovery ; run bootmenu_confirm_return
 bootmenu_4=Load production system via TFTP then write to NAND.=setenv noboot 1 ; setenv replacevol 1 ; run boot_tftp_production ; setenv noboot ; setenv replacevol ; run bootmenu_confirm_return
 bootmenu_5=Load recovery system via TFTP then write to NAND.=setenv noboot 1 ; setenv replacevol 1 ; run boot_tftp_recovery ; setenv noboot ; setenv replacevol ; run bootmenu_confirm_return
-bootmenu_6=[31mLoad BL31+U-Boot FIP via TFTP then write to NAND.[0m=run boot_tftp_write_fip ; run bootmenu_confirm_return
-bootmenu_7=[31mLoad BL2 preloader via TFTP then write to NAND.[0m=run boot_tftp_write_bl2 ; run bootmenu_confirm_return
-bootmenu_8=[31mInstall bootloader, recovery and production to eMMC.[0m=if mmc partconf 0 ; then run emmc_init ; else echo "eMMC not detected" ; fi ; run bootmenu_confirm_return
-bootmenu_9=Reboot.=reset
-bootmenu_10=Reset all settings to factory defaults.=run reset_factory ; reset
+bootmenu_6=[31mLoad BL31+U-Boot FIP via TFTP then write to NAND.[0m=run boot_tftp_write_fip ; run bootmenu_confirm_return
+bootmenu_7=[31mLoad BL2 preloader via TFTP then write to NAND.[0m=run boot_tftp_write_bl2 ; run bootmenu_confirm_return
+bootmenu_8=[31mInstall bootloader, recovery and production to eMMC.[0m=if mmc partconf 0 ; then run emmc_init ; else echo "eMMC not detected" ; fi ; run bootmenu_confirm_return
+bootmenu_9=[32mBoot production system from NVMe.[0m=run boot_nvme ; run bootmenu_confirm_return
+bootmenu_10=Reboot.=reset
+bootmenu_11=Reset all settings to factory defaults.=run reset_factory ; reset
 boot_first=if button reset ; then led $bootled_rec on ; run boot_tftp_recovery ; setenv flag_recover 1 ; run boot_default ; fi ; bootmenu
 boot_default=if env exists flag_recover ; then else run bootcmd ; fi ; run boot_recovery ; setenv replacevol 1 ; run boot_tftp_forever
 boot_production=led $bootled_pwr on ; run ubi_read_production && bootm $loadaddr#$bootconf#$bootconf_extra ; led $bootled_pwr off
 boot_recovery=led $bootled_rec on ; run ubi_read_recovery && bootm $loadaddr#$bootconf ; led $bootled_rec off
 boot_ubi=run boot_production ; run boot_recovery
+boot_nvme=pci enum ; nvme scan ; if nvme info ; then ext4load nvme 0:1 $loadaddr /bpi-r4.itb && bootm $loadaddr#$bootconf#mt7988a-bananapi-bpi-r4-nvme ; else echo NVMe not found ; fi
 boot_tftp_forever=led $bootled_rec on ; while true ; do run boot_tftp_recovery ; sleep 1 ; done
 boot_tftp_production=tftpboot $loadaddr $bootfile_upg && env exists replacevol && iminfo $loadaddr && run ubi_write_production ; if env exists noboot ; then else bootm $loadaddr#$bootconf#$bootconf_extra ; fi
 boot_tftp_recovery=tftpboot $loadaddr $bootfile && env exists replacevol && iminfo $loadaddr && run ubi_write_recovery ; if env exists noboot ; then else bootm $loadaddr#$bootconf ; fi
@@ -64,4 +66,4 @@
 _init_env=setenv _init_env ; run ubi_create_env ; saveenv ; saveenv
 _firstboot=setenv _firstboot ; run _switch_to_menu ; run _init_env ; run boot_first
 _switch_to_menu=setenv _switch_to_menu ; setenv bootdelay 3 ; setenv bootmenu_delay 3 ; setenv bootmenu_0 $bootmenu_0d ; setenv bootmenu_0d ; run _bootmenu_update_title
-_bootmenu_update_title=setenv _bootmenu_update_title ; setenv bootmenu_title "$bootmenu_title       [33m$ver[0m"
+_bootmenu_update_title=setenv _bootmenu_update_title ; setenv bootmenu_title "$bootmenu_title       [33m$ver[0m"
